src/arche/rules/result.py

```diff
@@ -80,6 +92,21 @@ def info(self):
     def warnings(self):
         return self.messages.get(Level.WARNING)
 
+    @property
+    def outcome(self) -> Outcome:
+        """
+        Returns the outcome status of a given rule
+        """
+        if not self.messages:
+            return Outcome.PASSED
+        message_level = self.messages.keys()
+        if any([level.value == Level.ERROR.value for level in message_level]):
+            return Outcome.FAILED
+        elif any([level.value == Level.WARNING.value for level in message_level]):
+            return Outcome.WARNING
+        else:
+            return Outcome.SKIPPED
```

Comment:
Docstring is obvious, let's just remove it.

The rest of the code could be simpler
```
if not self.messages:
    return Outcome.PASSED
 if Level.ERROR in self.messages.keys():
    return Outcome.FAILED
if Level.WARNING in self.messages.keys():
    return Outcome.WARNING
return Outcome.SKIPPED
```